src/callback.c

```diff
@@ -0,0 +1,19 @@
+/*
+ * Function pointers for callbacks for random number generators.
+ */
+double (*unif_rand_ptr)(void) = 0;
+double (*norm_rand_ptr)(void) = 0;
+double (*exp_rand_ptr)(void) = 0;
+
+double unif_rand(void)
+{
+    return (*unif_rand_ptr)();
+}
```

Comment:
My only remaining question is if it would be better if this were a header-defined `inline` function?